src/ucp/core/ucp_ep.h

```diff
@@ -176,4 +175,20 @@ ucp_md_map_t ucp_ep_config_get_rma_md_map(const ucp_ep_config_key_t *key,
 ucp_md_map_t ucp_ep_config_get_amo_md_map(const ucp_ep_config_key_t *key,
                                           ucp_lane_index_t lane);
 
+static UCS_F_ALWAYS_INLINE
+size_t ucp_ep_config_get_zcopy_auto_thresh(double iovcnt,
+                                           const uct_linear_growth_t *reg_cost,
+                                           double bcopy_bw, double bandwidth)
+{
+    double zcopy_thresh;
+
+    zcopy_thresh = (iovcnt * reg_cost->overhead) /
+                   ((1.0 / bcopy_bw) - (1.0 / bandwidth) - (iovcnt * reg_cost->growth));
+
+    if (zcopy_thresh < 0) {
+        return -1;
```

Comment:
Using `SIZE_MAX` required additional headers included into different files because this header used in C and C++ code. I decided to use `-1` to avoid additional files changes.
I will change it shortly.